File: src/vengine/scene/components/CameraComponent.cpp
#include <vengine/scene/components/CameraComponent.hpp>
#include "vengine/scene/Scene.hpp"
#include "vengine/scene/objects/SceneObject.hpp"
#include <glm/ext/matrix_transform.hpp>

namespace vengine::scene {

glm::mat4 CameraComponent::GetViewMatrix() const {
  const auto transform = GetWorldTransform();
  
  return inverse(transform.location.TranslationMatrix() * 
                   transform.rotation.Matrix());
}

glm::mat4 CameraComponent::GetProjection(const float aspectRatio) const {
  return glm::rotate(glm::perspective(glm::radians(fieldOfView), aspectRatio,  nearClipPlane,farClipPlane),glm::radians(180.f),{1,0,0});
}

void CameraComponent::Draw(drawing::SceneDrawer *drawer,
                           drawing::SimpleFrameData *frameData) {
  
}
}

File: rin.Scene/Components/CameraComponent.cs
using rin.Framework.Core.Math;
 
namespace rin.Scene.Components;

public class CameraComponent : RenderedComponent
{


    public float FieldOfView = 90.0f;

    public float NearClipPlane = 1f;

    public float FarClipPlane = 1000.0f;
    public Mat4 GetViewMatrix()
    {
        var worldTransform = GetWorldTransform();
        Mat4 transformMatrix = new Transform()
        {
            Location = worldTransform.Location,
            Rotation = worldTransform.Rotation
        };

        return transformMatrix.Inverse();
    }


    public Mat4 GetProjection(float width, float height)
    {
        return Glm.Perspective(float.DegreesToRadians(FieldOfView), width / height, NearClipPlane, FarClipPlane).RotateDeg(180.0f,new Vec3<float>(1f,0f,0f));
    }
}